orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,367 @@
+from AnyQt.QtWidgets import QApplication, QFormLayout, QTableView,  QSplitter
+from AnyQt.QtCore import Qt, QThread, pyqtSlot
+from PyQt5.QtGui import QDoubleValidator
+
+import numpy as np
+import scipy.stats as st
+
+import random
+import sys
+import copy
+import logging
+import concurrent.futures
+from functools import partial
+
+import Orange
+import Orange. evaluation
+from Orange.widgets.widget import OWWidget, Output, Input, Msg
+from Orange.widgets import gui, widget, settings
+from Orange.widgets.utils.itemmodels import TableModel
+from Orange.widgets.utils.sql import check_sql_input
+from Orange.base import Model
+from Orange.data import DiscreteVariable, ContinuousVariable, Domain, Table
+from Orange.widgets.utils.concurrent import (
+        ThreadExecutor, FutureWatcher, methodinvoke
+        )
+
+
+class Task:
+
+    future = ...
+    watcher = ...
+    canceled = False
+
+    def cancel(self):
+        self.canceled = True
+        self.future.cancel()
+        concurrent.futures.wait([self.future])
+
+class ExplainPredictions(object):
+    """
+    Class used to explain individual predictions by determining the importance of attribute values. All interactions between atributes are accounted for by calculating Shapely value.
+    Parameters:
+
+    :param data: table with dataset 
+    :type: Orange data table
+    :param model: model to be used for prediction
+    :type model: Orange model
+    :param pError: p value of error
+    :type pError: float
+    :param error: desired max error of approximation algorithm
+    :type error: float
+    :param batchSize: size of batch used in prediction, bigger batch speeds up the calculations
+    :type batchSize: int
+    :param maxIter: maximum number of iterations
+    :type maxIter: int
+    :param minIter: minimum number of iterations
+    :type minIter:int
```

Comment:
Docstrings are nice, but prefer [Numpydoc](https://numpydoc.readthedocs.io/en/latest/example.html) or [Google-style](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings as they are more readable in source.